Add batch.kubernetes.io/cronjob-scheduled-timestamp annotation information to cronjob page (#48498)
kubernetes/website: content/en/docs/concepts/workloads/controllers/cron-jobs.md

Kubernetes tries to avoid those situations, but does not completely prevent them. Therefore,
the Jobs that you define should be _idempotent_.

> Starting with Kubernetes v1.32 the job created by CronJob controller will have an annotation

Reviewer: ```suggestion
Starting with Kubernetes v1.32, the job created by CronJob controller has an annotation
```

Reviewer 2: Suggest: job --> Job

The CronJob is only responsible for creating Jobs that match its schedule, and
the Job in turn is responsible for the management of the Pods it represents.

> Starting with Kubernetes v1.32 the job created by CronJob controller will have an annotation `batch.kubernetes.io/cronjob-scheduled-timestamp` which will show the original (expected) creation timestamp for a Job. The value of this field is in RFC3339 format.

Reviewer 3: This is not technically true. Wouldn't this annotation have shown up in 1.28 and now we are forbidding users from turning it off in 1.32?

Author: Technically - yes, it should show up starting in 1.32, but since it was beta, it could potentially be disabled. Going with 1.32, where the feature is always on is safer assumption, b/c it will always appear. I thought for a while to go with 1.28, but based on what I wrote, I've decided to go with the stable.

> Starting with Kubernetes v1.32 the job created by CronJob controller will have an annotation `batch.kubernetes.io/cronjob-scheduled-timestamp` which will show the original (expected) creation timestamp for a Job. The value of this field is in RFC3339 format.

Reviewer 4: I think it is a bit confusing to place this after:

```
The CronJob is only responsible for creating Jobs that match its schedule, and
the Job in turn is responsible for the management of the Pods it represents.
```

Can we either:
- highlight here why this annotation is useful? For example: `can be useful if the job has not been created according to the schedule`
- or just place above the last sentence or somewhere else in the Job creation section?

Author: Good point, moved it up, after the first paragraph.
